xonsh/winutils.py

```diff
@@ -249,6 +249,14 @@ def set_console_mode(mode, fd=1):
     SetConsoleMode(hcon, mode)
 
 
+def enable_virtual_terminal_processing():
+    """Enables virtual terminal processing on Windows.
+    This inlcudes ANSI escape sequence interpretation.
+    See http://stackoverflow.com/a/36760881/2312428
+    """
+    SetConsoleMode(GetStdHandle(-11), 7)
```

Comment:
> Shouldn't we set this for both the readline shell and the PTK shell? Just so ANSI out is always handled correctly.

I think we can do that easily if we find any cases that fail.  In general, though, it seems like PTK handles all of this stuff internally for us. PTK does set the console mode in a much a places.

> Also, I think this can fail if we are on old versions of windows. In that case we can assume that ConHost doesn't have the capabilities.

So right now, if it fails it is just a silent failure.  We could put in a warning in debug mode if you think it is needed.